### PR TITLE
Issues/51

### LINKED ISSUE #51: Recording video from app
There is no option to record video right now, research need to be done, to find best fitting technology:
- [x] OpenCV - record video
- [x] QtGui.QPixmap.grabWindow() - convert slides to video
- [x] QWidget.grab() - previous is deprecated in Qt5.

### PR DESCRIPTION
**Pull request by LemurPwned**
Resolves #51 
although I dont think this will work without external dependencies like ffmpeg

### CODE REVIEW

File: Widgets/openGL_widgets/AbstractGLContext.py [hunk 1]
```diff
@@ -38,6 +40,35 @@ def handleOptionalData(self):
         except NameError:
             self.iterations = 1
 
+    def auto_center(self):
```
Comment (pawelkulig1) [inline Widgets/openGL_widgets/AbstractGLContext.py]: I think moving this here have benefit of not repeating code, but readability is lost a little.

File: Widgets/plot_widgets/Canvas2Dupgraded.py [hunk 2]
```diff
@@ -1,41 +1,41 @@
 import numpy as np
 from pyqtgraph import PlotWidget
 import pyqtgraph as pg
-from AnimatedWidget import AnimatedWidget #can inherit but \
-# have to overwrite anyway
+from AnimatedWidget import AnimatedWidget
 
 class Canvas2Dupgraded(PlotWidget, AnimatedWidget):
-        def __init__(self, parent=None, data_dict=None):
```
Comment (pawelkulig1) [inline Widgets/plot_widgets/Canvas2Dupgraded.py]: why there was double tab? 😮

File: main.py [hunk 3]
```diff
@@ -47,6 +50,7 @@ def __init__(self):
         self.make1WindowGrid()  # shows default 1 widget Window
         self.events()  # create event listeners
         self._LOADED_FLAG_ = False
+        self.screenshot_dir = 'Screenshots'
```
Comment (pawelkulig1) [inline main.py]: Shouldn't be hardcoded, but for beta release it's ok.

File: main.py [hunk 4]
```diff
@@ -129,28 +161,21 @@ def loadFile(self):
 
     def loadDirectory(self):
         """Loads whole directory based on Parse class as simple as BHP"""
-        fileDialog = QtWidgets.QFileDialog()
```
Comment (pawelkulig1) [inline main.py]: I don't think this kind of refactoring is necessary.

File: Widgets/plot_widgets/Canvas2Dupgraded.py (hunk 2 above)
Comment (LemurPwned) [reply Widgets/plot_widgets/Canvas2Dupgraded.py]: I don't know I use single indents

File: main.py (hunk 4 above)
Comment (LemurPwned) [reply main.py]: file is a keyword in Python. It should not be used as a variable

Comment (pawelkulig1) [reply main.py]: file - yes, fileDialog - no

Comment (LemurPwned) [reply main.py]: it's just code reusability

**Pull request merged**